contingency/contingency-api/src/main/java/com/powsybl/contingency/json/ContingencyElementDeserializer.java

```diff
@@ -58,6 +58,7 @@ public ContingencyElement deserialize(JsonParser parser, DeserializationContext
                 case LOAD -> new LoadContingency(id);
                 case BUS -> new BusContingency(id);
                 case TIE_LINE -> new TieLineContingency(id, voltageLevelId);
+                case BATTERY -> new BatteryContingency(id);
```

Comment:
Seen with @EtienneLt: Switch contingencies are not fully supported.  
This will be the object of another PR.